### ISSUE TITLE
About the gap between testing result and validation result on SemanticKITTI

### ISSUE THREAD

**Issue by liamlin5566** [PRBonn/4DMOS#24]
Hi, thanks for sharing your great work.
I am confused with that your validation result is 77.2% shown in paper while your testing result is just 65.2%. Could you explain the gap between testing and validation result? Some issues like #5 indicate that the pose affects the performance significantly. Is it the reason resulting in this gap? Thanks.

**Comment by benemer** [comment]
Hey,

this is hard to say without having the labels at hand, since the benchmark only reports the IoU on the test set and for example no Precision/Recall metrics.

Besides the already mentioned influence of drifting poses, another reason could be the labeling strategy of SemanticKITTI. Since an object is considered moving if it has moved throughout the sequence, cars standing in a traffic like in test sequence 20 will have the ground truth label `moving` whereas 4DMOS might only see the cars standing still within the window of past scans.

Also, be aware that during training, we pick the checkpoint of the model with the highest IoU on the validation set. So it is expected to have a very good result here and a possible performance gap to sequences from the unseen test set. This gap can also be observed with other MOS methods, see [LMNet](https://github.com/PRBonn/LiDAR-MOS) and [MotionSeg3D](https://github.com/haomo-ai/MotionSeg3D).

**Comment by liamlin5566** [comment]
Thanks for replying. I wonder whether you also use the poses without loop closing for training to obtain better result.

**Comment by liamlin5566** [comment]
BTW, I noticed that you use KITTI-Road dataset to train and get better testing result. Do you use all KITTI-Road data for training or follow the dataset splitting rule from MotionSeg3D (https://github.com/haomo-ai/MotionSeg3D/blob/master/config/labels/semantic-kitti-mos.yaml).

**Comment by benemer** [comment]
> Thanks for replying. I wonder whether you also use the poses without loop closing for training to obtain better result.

No we did not, because we wanted to have exactly the same training setup as other methods.



> BTW, I noticed that you use KITTI-Road dataset to train and get better testing result. Do you use all KITTI-Road data for training or follow the dataset splitting rule from MotionSeg3D (https://github.com/haomo-ai/MotionSeg3D/blob/master/config/labels/semantic-kitti-mos.yaml).

We added all labeled road sequences from [here](https://github.com/haomo-ai/MotionSeg3D/blob/master/config/kitti_road_mos.md) to the training set and obtained better results on the test set. However, we later realized that the road sequence 41 is a subset of test sequence 20. So road sequence 41 should be neither in the training nor in the validation set. I removed this entry from the benchmark. We did not train any other setup so far.

**Comment by liamlin5566** [comment]
OK, thanks for answering my questions.
